src/accelerate/accelerator.py

```diff
@@ -155,6 +160,16 @@ def __init__(
                 deepspeed_plugin, DeepSpeedPlugin
             ), "`deepspeed_plugin` must be a DeepSpeedPlugin object."
 
+        if os.environ.get("USE_FSDP", "false") == "true":
+            if version.parse(torch.__version__) < version.parse("1.12.0.dev20220418+cu113"):
+                raise ValueError("FSDP requires PyTorch >= 1.12.0.dev20220418+cu113")
+            if fsdp_plugin is None:  # init from env variables
+                fsdp_plugin = FullyShardedDataParallelPlugin()
+            else:
+                assert isinstance(
+                    fsdp_plugin, FullyShardedDataParallelPlugin
+                ), "`fsdp_plugin` must be a FullyShardedDataParallelPlugin object."
```

Comment:
For new code, we try to avoid assert, so please use a test and rasie a `TypeError` here.

Reply:
Done.